checkov/terraform/graph_builder/local_graph.py

```diff
@@ -311,17 +315,17 @@ def _get_target_variables(self, vertex: TerraformBlock, dest_module_path: str) -
             ]
 
     def _build_cross_variable_edges(self):
-        target_nodes_indexes = [v for v, referenced_vertices in self.out_edges.items() if
-                                self.vertices[v].block_type == BlockType.RESOURCE and any(
-            self.vertices[e.dest].block_type != BlockType.RESOURCE for e in referenced_vertices)]
         aliases = self._get_aliases()
         resources_types = self.get_resources_types_in_graph()
-        for origin_node_index in target_nodes_indexes:
-            vertex = self.vertices[origin_node_index]
-            self._build_edges_for_vertex(origin_node_index, vertex, aliases, resources_types, True)
-            modules = vertex.breadcrumbs.get(CustomAttributes.SOURCE_MODULE, [])
-            for module in modules:
-                self._build_edges_for_vertex(origin_node_index, vertex, aliases, resources_types, True, module)
+        for v, referenced_vertices in self.out_edges.items():
+            if self.vertices[v].block_type == BlockType.RESOURCE and \
+                    any(self.vertices[e.dest].block_type != BlockType.RESOURCE for e in referenced_vertices):
+                origin_node_index = v
```

Comment:
maybe I'm missing something, but I don't see the reason for the extra variable here. How about changing the loop variable `v` to `origin_node_index` in the for_loop directly?

Reply:
you are right, it's a leftover from a check I made